project/Dependencies.scala

```diff
@@ -53,30 +70,24 @@ object Dependencies {
     )
   }
 
+  def agentDependencies(sparkVersion: String): Seq[ModuleID] = {
 
-  val agentDependencies: Seq[ModuleID] = {
-
-    val spark3Version = "3.3.2"
-    val scala212 = "2.12.12"
-    val scalatestVersion = "3.2.15"
-    val specs2Version = "4.19.2"
-    val typesafeConfigVersion = "1.4.2"
-
-    lazy val sparkCore = "org.apache.spark" %% "spark-core" % spark3Version
-
-    lazy val sparkCommons = "za.co.absa" % "spark-commons-spark3.3_2.12" % "0.5.0"
-
-    lazy val sparkCommonsTest = "za.co.absa" %% "spark-commons-test" % "0.5.0" % Test
-
-    lazy val sparkSql ="org.apache.spark" %% "spark-sql" %  spark3Version
-    lazy val scalaTest = "org.scalatest" %% "scalatest" % scalatestVersion % Test
-    lazy val specs2core = "org.specs2" %% "specs2-core" % specs2Version % Test
-    lazy val typeSafeConfig = "com.typesafe" % "config" % typesafeConfigVersion
+    val typesafeVersion     = "1.4.2"
+    val sparkCommonsVersion = "0.6.0"
+    val sparkMinorVersion   = getVersionUpToMinor(sparkVersion)
+    val specs2CoreVersion   = "4.19.2"
 
+    Seq(
+        "org.apache.spark" %% "spark-core"                               % sparkVersion % Provided,
+        "org.apache.spark" %% "spark-sql"                                % sparkVersion % Provided,
+        "com.typesafe"      % "config"                                   % typesafeVersion, //TODO to check if used
```

Comment:
Found TODO in this file.
In the code there are 4 another TODOs - these should be updated with Issue where is planned to solved them.